exporters/otlp/test/otlp_http_exporter_test.cc

```diff
@@ -124,18 +126,22 @@ class OtlpHttpExporterTestPeer : public ::testing::Test, public HTTP_SERVER_NS::
       else
       {
         response.body = "{\"code\": 400, \"message\": \"Unsupported content type\"}";
-        return 400;
+        response_status = 400;
       }
 
-      return 200;
+      response_status = 200;
     }
     else
     {
       std::unique_lock<std::mutex> lk(mtx_requests);
       response.headers["Content-Type"] = "text/plain";
       response.body                    = "404 Not Found";
-      return 200;
+      response_status = 200;
     }
+
+    cv_got_events.notify_one();
```

Comment:
why is this needed ? `cv.wait_for` was supposed to be unblocked with a timeout. I think this will fail now if the test is modified so that `cv.wait_for` has multiple requests to wait for, as now it will be unblocked after a single request?

Reply:
Timeout in `cv.wait_for` is the last resort. `cv` should be waked up anytime its checking condition could be changed, so its condition could be re-checked instead of waiting for timing out. `notify_one` just tell `cv` to recheck so it should not trigger failure behavior?

cc @owent

Reply:
Does it mean that without notification, the `cv` will keep blocked for 2sec (as configured) even though the condition is satisfied early?

Reply:
Yes, this is the case the PR trying to fix.

Reply:
Asking because we are using this construct in other places too (https://github.com/open-telemetry/opentelemetry-cpp/blob/7fca2c39586696d71de4f1feeb9ee6705a4e0261/ext/test/http/curl_http_test.cc#L124), probably they need a change too ( in separate PR ).

Reply:
You are right, it is needed in curl_http_test.cc as well. Added the change to the PR and avoid another 2 timeout to happen there. Thanks for the catch.

Reply:
> Timeout in `cv.wait_for` is the last resort. `cv` should be waked up anytime its checking condition could be changed, so its condition could be re-checked instead of waiting for timing out. `notify_one` just tell `cv` to recheck so it should not trigger failure behavior?
> 
> cc @owent

You are right. LGTM.